src/puppetlabs/puppetdb/admin.clj

```diff
@@ -12,12 +14,41 @@
             [puppetlabs.puppetdb.middleware :as mid]
             [bidi.schema :as bidi-schema]
             [puppetlabs.puppetdb.schema :as pls]
-            [schema.core :as s]))
+            [schema.core :as s]
+            [slingshot.slingshot :refer [try+]]))
+
+(def clean-command-schema
+  {:version (s/eq 1)
+   :payload [(apply s/enum (map name cli-svc/clean-options))]})
+
+(def ^:private validate-clean-command (s/validator clean-command-schema))
+
+(defn- handle-clean-req [req clean]
+  (let [body (slurp (:body req))
+        command (json/parse-string body true)
+        invalid-info (try+
+                      (validate-clean-command command)
+                      nil
+                      (catch [:type :schema.core/error] ex
+                        ex))]
+    (cond
+      invalid-info
+      (http/bad-request-response (format "Invalid command %s: %s"
+                                         (pr-str body)
+                                         (pr-str (:error invalid-info))))
+
+      (deref (future (clean (map keyword (:payload command)))))
+      (http/json-response {:ok true})
+
+      :else
+      (http/json-response {:info "Another cleanup is already in progress"}
+                          http/status-conflict))))
```

Comment:
That's exactly what I was looking for @kbarber I couldn't find that in confluence